source/base/data_out_base.cc

```diff
@@ -7544,6 +7544,80 @@ DataOutInterface<dim, spacedim>::write_pvtu_record(
 }
 
 
+template <int dim, int spacedim>
+void
+DataOutInterface<dim, spacedim>::write_vtu_with_pvtu_record(
+  const std::string &folder,
+  const std::string &filename_without_extension,
+  const unsigned int counter,
+  const unsigned int n_digits_counter,
+  const MPI_Comm &   mpi_communicator,
+  const unsigned int n_groups) const
+{
+  const unsigned int rank = Utilities::MPI::this_mpi_process(mpi_communicator);
+  const unsigned int n_ranks =
+    Utilities::MPI::n_mpi_processes(mpi_communicator);
+  const unsigned int n_files_written =
+    (n_groups == 0 || n_groups > n_ranks) ? n_ranks : n_groups;
+
+  const unsigned int n_digits =
+    std::ceil(std::log10(std::fabs(n_files_written)));
+
+  const unsigned int color =
+    (n_groups == 0 || n_groups > n_ranks) ? rank : rank % n_groups;
+  std::string processor_id =
+    (n_files_written > 1) ? ("." + Utilities::int_to_string(color, n_digits)) :
+                            "";
+  std::string filename = folder + filename_without_extension + "_" +
+                         Utilities::int_to_string(counter, n_digits_counter) +
+                         processor_id + ".vtu";
+  std::ofstream output(filename.c_str());
+
+  // every processor writes one file
+  if (n_groups == 0 || n_groups > n_ranks)
+    {
+      this->write_vtu(output);
+    }
+  // write only a single data file in parallel
+  else if (n_groups == 1)
+    {
+      this->write_vtu_in_parallel(filename.c_str(), mpi_communicator);
+    }
+  // write n_groups data files
+  else
+    {
+      MPI_Comm comm_group;
+      MPI_Comm_split(mpi_communicator, color, rank, &comm_group);
+      this->write_vtu_in_parallel(filename.c_str(), comm_group);
+      MPI_Comm_free(&comm_group);
+    }
+
+  // pvtu record only needed if more than one file is written
+  if (n_files_written > 1)
+    {
+      if (rank == 0)
+        {
+          std::vector<std::string> vector;
```

Comment:
please don't call this variable ``vector``

Reply:
done in https://github.com/dealii/dealii/pull/8904/commits/f07a8bf9bce24520305b079bc179599d78782753